src/compass/s1_geocode_slc.py

```diff
@@ -167,68 +157,120 @@ def run(cfg: GeoRunConfig):
-                rdr_burst_raster = isce3.io.Raster(temp_slc_path)
-
-                # Declare names, types, and descriptions of respective outputs
-                ds_names = [pol, 'azimuth_carrier_phase', 'flattening_phase']
-                ds_types = ['complex64', 'float64', 'float64']
-                ds_descrs = [f'{pol} geocoded CSLC image{desc}'
-                             for desc in ['', ' azimuth carrier phase',
-                                          ' flattening phase']]
-
-                # Iterate over zipped names, types, and descriptions and create
-                # raster objects
-                geo_burst_raster, carrier_raster, flatten_phase_raster =\
-                    [_init_geocoded_IH5_raster(grid_group, ds_name, geo_grid,
-                                               ds_type, ds_desc)
-                     for ds_name, ds_type, ds_desc in zip(ds_names, ds_types,
-                                                          ds_descrs)]
-
-                # Geocode
-                geocode_slc(geo_burst_raster, rdr_burst_raster, dem_raster,
-                            radar_grid, sliced_radar_grid, geo_grid, orbit,
-                            native_doppler, image_grid_doppler, ellipsoid,
-                            threshold, iters, blocksize, flatten, reramp=True,
-                            azimuth_carrier=az_carrier_poly2d,
-                            az_time_correction=az_lut,
-                            srange_correction=rg_lut,
-                            carrier_phase_raster=carrier_raster,
-                            flatten_phase_raster=flatten_phase_raster)
-
-            # Set geo transformation
-            geotransform = [geo_grid.start_x, geo_grid.spacing_x, 0,
-                            geo_grid.start_y, 0, geo_grid.spacing_y]
-            geo_burst_raster.set_geotransform(geotransform)
-            geo_burst_raster.set_epsg(epsg)
-
-            # ISCE3 raster IH5 cleanup
-            del geo_burst_raster
-            del carrier_raster
-            del flatten_phase_raster
+                # Load input dataset of current polarization as array from GDAL
+                # raster
+                rdr_dataset = gdal.Open(temp_slc_path, gdal.GA_ReadOnly)
+                rdr_data_blks.append(rdr_dataset.ReadAsArray())
+
+                # Prepare output dataset of current polarization in HDF5
+                geo_ds = init_geocoded_dataset(grid_group, pol, geo_grid,
+                                               'complex64',
+                                               f'{pol} geocoded CSLC image',
+                                               output_cfg=cfg.output_params)
+                geo_datasets.append(geo_ds)
+
+                # Init geocoded output blocks/arrays lists to NaN
+                geo_data_blks.append(
+                    np.full(out_shape, np.nan + 1j * np.nan).astype(np.complex64))
+
+            dt_prep = get_time_delta_str(t_prep)
```

Comment:
side note:
```
journal: s1_geocode_slc geocode prep time 0:00:32 (hr:min:sec)
```
this seems kinda slow for just loading data? weird, but we'll look later at it